### PR TITLE
The C++ Inference Library of OpenPose-Plus v2.0.0

### PR DESCRIPTION
**Pull request by ganler**
## Features

- New abstraction with better performance
    - `Operator API`: For users to manage detailed DNN inference and post-processing. (\~90 FPS in video processing)
    - `Stream API`: Based on `Operator API`, we made a scheduler to do stream processing, provide end2end video processing. You can have it with very simple C++ stream operators like `<<` or `>>` to set the input/output streams. (\~120FPS in video processing).
- ONNX model parser support.
- Documentation.

The performance was tested on 1070 Ti with 6 physical core CPU.

## WIP

- Documentation.
    - [x] Doxygen Comments.
    - [x] Doxygen to Sphinx ~~via Breathe~~.
    - [ ] Deployment.
- Post-Processing.
    - [ ] Proposal Network(NMS).
- Benchmark over new models.
    - [x] OpenPose model.
    - [x] Lightweight OpenPose.
    - [ ] Proposal Network
- MISC.
    - [x] Tests on ONNX parser.
    - [x] Add serialized TensorRT Engine file.
    - [x] Fix merge conflicts
    - [ ] CIs.
    - [ ] Ensure the scripts and docker running.
    - [ ] Provide a end2end simple docker application to generate a processed video.

Looking for preliminary reviews and contributions.

### CODE REVIEW

File: include/openpose_plus/utility/data.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#include "viz.hpp"
+
+#include <opencv2/opencv.hpp>
+#include <ttl/tensor>
+#include <vector>
+
+namespace poseplus {
+
+template <typename... MatType>
+std::vector<cv::Mat> make_batch(MatType... mats)
```
Comment (lgarithm) [inline include/openpose_plus/utility/data.hpp:12]: make_batch is not used.

File: include/openpose_plus/operator/dnn/tensorrt.hpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+#include <future>
+
+#include <ttl/cuda_tensor>
+
+#include "../../utility/data.hpp"
+
+namespace poseplus {
+
+/// Element data type.
+using data_type = nvinfer1::DataType;
```
Comment (lgarithm) [inline include/openpose_plus/operator/dnn/tensorrt.hpp:19]: alias is not used

File: include/openpose_plus/utility/data.hpp (hunk 1 above)
Comment (ganler) [reply include/openpose_plus/utility/data.hpp:12]: I'll remove it. :-)

File: include/openpose_plus/operator/dnn/tensorrt.hpp (hunk 2 above)
Comment (ganler) [reply include/openpose_plus/operator/dnn/tensorrt.hpp:19]: Yes. But I don't really like users to touch types from other libraries(e.g., `nvinferN::XXX`). 
And it's meaningful when we support other engines. What do you think?

**Comment by lgarithm** [comment]
please resolve conflicting files.

**Comment by ganler** [comment]
The documentation comments and conflicts are solved. 
Do you have some time to look at the Travis CI? It seems to be a problem with CUDA.

**Comment by ganler** [comment]
@lgarithm I saw your comments about OpenCV on the CI scripts. Is there a problem when building OpenCV in docker? Is this helpful? https://hub.docker.com/r/schickling/opencv

**Comment by lgarithm** [comment]
The CI is failing because CUDA become a required dependency.
We used to have a stub (fake_openpose_plus.cpp) for testing purpose, which is now removed.

**Comment by lgarithm** [comment]
There are two options to fix:
1). use container with tensorRT and CUDA for CI.
2). revert `fake_openpose_plus.cpp`

**Comment by ganler** [comment]
@lgarithm I think for the best, we'd better select option 1. Because "fake_openpose_plus" may not simulate the real situation of compiling on GPU. Can we get this done by using `nvidia-docker`? (I'm not very familiar with docker. :(

**Comment by ganler** [comment]
I removed `fake_openpose_plus` because the older APIs are deprecated.

**Comment by ganler** [comment]
Thanks to @Gyx-One , we successfully converted the official OpenPose model to ONNX format which enables TensorRT execution. We firstly got 20 fps by inferring 20 batched images.

**Comment by lgarithm** [comment]
@ganler could you make files end with '\n'

**Pull request merged**